roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -198,7 +209,17 @@ def is_available_skopeo_image(self, image, default_registries):
             if not self.reachable_registries[registry]:
                 continue
 
-            args = {"_raw_params": self.skopeo_img_check_command.format(registry=registry, image=image)}
+            skopeo_command_kwargs = {
+                'registry': registry,
+                'image': image
+            }
+            if self.oreg_auth_user is not None:
+                # skopeo should complete without complaining if credentials are
```

Comment:
I have thought about this, and I'm uncomfortable shipping as-is.

We need a way to ensure we're only sending our credentials to the correct host.  We should probably also enable tls-verify for this by default, unless the user specifically opts-out.

Sending credentials to un-verified registries, potentially due to a type like 'dokcer.io' or something similar, is a bad practice.

Need input here.  Should we handle oreg_url separately from the rest of the registries?

Reply:
> We should probably also enable tls-verify for this by default, unless the user specifically opts-out.

👍 ... Unless the registry is in the insecure list we should assume TLS. Maybe the ``skopeo_img_check_command`` should be ```"timeout 10 skopeo inspect --tls-verify={tls} docker://{registry}/{image}"``` where ``tls`` becomes false only if the registry is in the insecure registry list.

> Sending credentials to un-verified registries, potentially due to a type like 'dokcer.io' or something similar, is a bad practice.

Agreed.

> Need input here. Should we handle oreg_url separately from the rest of the registries?

I'm not familiar enough with what ``oreg_url`` is used for to give good guidance. @sosiouxme ideas?

Reply:
Good catch; it wasn't a concern before we started sending anything sensitive. I think @ashcrow's suggestion should work though. I can work on tweaking and testing if you'd prefer.

Reply:
For that matter, it's probably not OK to send credentials for your private repo to docker.io, TLS or no.

`oreg_auth_*` doesn't give any method for indicating different authentication for different registries.

Maybe assume it only applies to additional repos? Until we provide a way to be more specific.

Reply:
@sosiouxme that's what I mean, it's probably not okay to send a cert, even if tls is valid, for a myriad of reasons, eg, typo in additiona_registries could post credentials to a malicious actor; It's generally a bad practice to send credentials to the wrong systems, even if trusted, perhaps safe-registry.com is logging passwords unencrypted somewhere.

oreg_auth_* is only designed to work with oreg_url's host.  In the docker/node/master roles, we're running docker login only against oreg_host (derived from oreg_url).

We can probably just work in some code to only use auth against oreg, if necessary.